model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/JpaMapStorageProviderFactory.java

```diff
@@ -265,6 +265,7 @@ public void init(Config.Scope config) {
 
     @Override
     public void postInit(KeycloakSessionFactory factory) {
+        this.emf = createEntityManagerFactory();
     }
 
```

Comment:
When testing this in Undertow and Quarkus, I found that this is being called twice (actually the logging in the undertow setup put me on this). 

Due to this, at least in the Undertow setup two EMFs will be set up, with two connection pools. I assume the same happens for Quarkus as well.

I remember this happens for all (?) `AmphibianProviderFactory`. This was apparently the reason why it was lazily initialized. 

Locally I rolled back the change and made it lazily-initializable. This then still worked for both Undertow and Quarkus.


```suggestion
    @Override
    public void postInit(KeycloakSessionFactory factory) {
    }

    // ... and then call lazyInit() again from create()
    private void lazyInit() {
        if (emf == null) {
            synchronized (this) {
                if (emf == null) {
                    this.emf = createEntityManagerFactory();
                }
            }
        }
    }